lib/onelogin/ruby-saml/utils.rb

```diff
@@ -1,44 +1,83 @@
+      # @param heads [Boolean] If true, the formatted certificate will include the
+      #                        "BEGIN CERTIFICATE" header and the footer.
+      # @return [String] The formatted certificate
+      #
       def self.format_cert(cert, heads=true)
-        cert = cert.delete("\n").delete("\r").delete("\x0D")
-        if cert
-          cert = cert.gsub('-----BEGIN CERTIFICATE-----', '')
-          cert = cert.gsub('-----END CERTIFICATE-----', '')
-          cert = cert.gsub(' ', '')
-
+        unless cert.empty?
+          cert = cert.gsub(/\-{5}\s?(BEGIN|END) CERTIFICATE\s?\-{5}/, "")
+          cert = cert.gsub(/[\n\r\s]/, "")
+          cert = cert.scan(/.{1,64}/).join("\n")+"\n"
           if heads
-            cert = cert.scan(/.{1,64}/).join("\n")+"\n"
             cert = "-----BEGIN CERTIFICATE-----\n" + cert + "-----END CERTIFICATE-----\n"
           end
         end
         cert
       end
 
+      # Return the private key string formatted
+      # @param key [String] The original private key 
+      # @param heads [Boolean] If true, the formatted private key will include the
+      #                  "BEGIN PRIVATE KEY" or the "BEGIN RSA PRIVATE KEY" header and the footer.
+      # @return [String] The formatted certificate
+      #
       def self.format_private_key(key, heads=true)
-        key = key.delete("\n").delete("\r").delete("\x0D")
-        if key
+        unless key.empty?
+          key = key.delete("\n").delete("\r").delete("\x0D")
           if key.index('-----BEGIN PRIVATE KEY-----') != nil
             key = key.gsub('-----BEGIN PRIVATE KEY-----', '')
             key = key.gsub('-----END PRIVATE KEY-----', '')
             key = key.gsub(' ', '')
+
+            key = key.scan(/.{1,64}/).join("\n")+"\n"
             if heads
-              key = key.scan(/.{1,64}/).join("\n")+"\n"
               key = "-----BEGIN PRIVATE KEY-----\n" + key + "-----END PRIVATE KEY-----\n"
             end
           else
             key = key.gsub('-----BEGIN RSA PRIVATE KEY-----', '')
             key = key.gsub('-----END RSA PRIVATE KEY-----', '')
             key = key.gsub(' ', '')
+            key = key.scan(/.{1,64}/).join("\n")+"\n"
             if heads
-              key = key.scan(/.{1,64}/).join("\n")+"\n"
               key = "-----BEGIN RSA PRIVATE KEY-----\n" + key + "-----END RSA PRIVATE KEY-----\n"
             end
           end
         end
+        key
       end
 
+      # Build the Query String signature that will be used in the HTTP-Redirect binding
+      # to generate the Signature
+      # @param type        [String] 'SAMLRequest' or 'SAMLResponse'
+      # @param data        [String] The plain text request or response
+      # @param relay_state [String] The RelayState parameter
+      # @param sig_alg     [String] The SigAlg parameter                  
+      # @return [String] The Query String 
+      #
+      def self.build_query(type, data, relay_state, sig_alg)
```

Comment:
If we're accepting this many parameters, my 5 cents would be to use an options hash so when we call `Utils.build_query` it's readable to which params we're sending, besides, the order of the params wouldn't matter.